src/member-api.js

```diff
@@ -51,19 +56,38 @@ export class MemberApi extends TypedEmitter {
    *
    * @param {Object} opts
    * @param {import('./capabilities.js').RoleId} opts.roleId
+   * @param {string} [opts.roleName]
+   * @param {string} [opts.roleDescription]
    * @param {number} [opts.timeout]
    *
    * @returns {Promise<import('./generated/rpc.js').InviteResponse_Decision>}
    */
-  async invite(deviceId, { roleId, timeout }) {
-    const projectId = projectKeyToId(this.#projectKey)
+  async invite(deviceId, { roleId, roleName, roleDescription, timeout }) {
+    if (!(roleId === MEMBER_ROLE_ID || roleId === COORDINATOR_ROLE_ID)) {
+      throw new Error('Invalid role id')
+    }
+
+    const { name: deviceName } = await this.getById(this.#ownDeviceId)
+
+    // since we are always getting #ownDeviceId,
+    // this should never throw (see comment on getById), but it pleases ts
+    if (!deviceName) throw new Error('Invalid deviceName')
```

Comment:
```suggestion
    if (!deviceName) throw new Error('Internal error trying to read own device name for this invite')
```

Trying to make the error message a bit more descriptive, since the current one is misleading because `deviceName` is not a parameter for this function.